apps/judicial-system/backend/src/app/modules/notification/test/notificationController/sendReadyForCourtNotifications.spec.ts

```diff
@@ -0,0 +1,233 @@
+  const courtCaseNumber = uuid()
+  const theCase = {
+    id: caseId,
+    type: CaseType.CUSTODY,
+    state: CaseState.RECEIVED,
+    policeCaseNumbers: [policeCaseNumber],
+    prosecutor: {
+      name: 'Derrick',
+      email: 'derrick@dummy.is',
+      institution: { name: 'Héraðsdómur Derricks' },
+    },
+    courtId,
+    court: { name: 'Héraðsdómur Reykjavíkur' },
+    courtCaseNumber,
+    courtDate: randomDate(),
+    defenderName: 'Saul Goodman',
+    defenderEmail: 'saul@dummy.is',
+    sendRequestToDefender: true,
+  } as Case
+  const user = { id: uuid() } as User
+  const notification = { type: NotificationType.READY_FOR_COURT }
+  const now = randomDate()
+  const courtMobileNumber = uuid()
+  let mockEmailService: EmailService
+  let mockSmsService: SmsService
+  let mockCourtService: CourtService
+  let mockNotificationModel: typeof Notification
+  let givenWhenThen: GivenWhenThen
+
+  beforeEach(async () => {
+    process.env.COURTS_MOBILE_NUMBERS = `{"${courtId}": "${courtMobileNumber}"}`
+
+    const mockNowFactory = nowFactory as jest.Mock
+    mockNowFactory.mockReturnValue(now)
+
+    const {
+      emailService,
+      smsService,
+      courtService,
+      notificationModel,
+      notificationController,
+    } = await createTestingNotificationModule()
+
+    mockEmailService = emailService
+    mockSmsService = smsService
+    mockCourtService = courtService
+    mockNotificationModel = notificationModel
+
+    givenWhenThen = async (caseId, user, theCase, notification) => {
+      const then = {} as Then
+
+      await notificationController
+        .sendCaseNotification(caseId, user, theCase, notification)
+        .then((result) => (then.result = result))
+        .catch((error) => (then.error = error))
+
+      return then
+    }
+  })
+
+  describe('first notification', () => {
+    let then: Then
+
+    beforeEach(async () => {
+      then = await givenWhenThen(caseId, user, theCase, notification)
+    })
+
+    it('should lookup previous ready for court notifications', () => {
+      expect(mockNotificationModel.findOne).toHaveBeenCalledWith({
+        where: { caseId, type: NotificationType.READY_FOR_COURT },
+      })
+    })
+
+    it('should send ready for court email notification to prosecutor', () => {
+      expect(mockEmailService.sendEmail).toHaveBeenCalledWith({
+        from: { name: 'Réttarvörslugátt', address: 'ben10@omnitrix.is' },
+        replyTo: { name: 'Réttarvörslugátt', address: 'ben10@omnitrix.is' },
+        to: [{ name: 'Derrick', address: 'derrick@dummy.is' }],
+        subject: 'Krafa um gæsluvarðhald send',
+        text: `Þú hefur sent kröfu á Héraðsdóm Reykjavíkur vegna LÖKE máls ${policeCaseNumber}. Skjalið er aðgengilegt undir málinu í Réttarvörslugátt.`,
+        html: `Þú hefur sent kröfu á Héraðsdóm Reykjavíkur vegna LÖKE máls ${policeCaseNumber}. Skjalið er aðgengilegt undir <a href="http://localhost:4200/krafa/stadfesta/${caseId}">málinu í Réttarvörslugátt</a>.`,
```

Comment:
Consider taking `http://localhost:4200` from the config.

Reply:
I guess the test would be more robust by simply referring to the config rather than hard coding the text. I will make that change.

This will not remove `http://localhost:4200` from the config however.

Reply:
Done